### ISSUE TITLE
route53_record alias keeps being recreated when elb dns_name is not entirely lowercase

### ISSUE THREAD

**Issue by blckct** [terraform-providers/terraform-provider-aws#1005]
### Terraform Version
v0.9.8

### Affected Resource(s)
- aws_route53_record
- aws_elb.

### Terraform Configuration Files
```hcl
resource "aws_route53_zone" "internal" {
  name = "elbtest.deleteme"
}

resource "aws_elb" "lb" {

  availability_zones=["eu-west-1a"]
  name = "Test"

  "listener" {
    instance_port = 80
    instance_protocol = "HTTP"
    lb_port = 80
    lb_protocol = "HTTP"
  }
}

// Attach the ELB to domain
resource "aws_route53_record" "elb" {
  name = "test.elbtest.deleteme"
  type = "A"
  zone_id = "${aws_route53_zone.internal.zone_id}"
  alias {
    evaluate_target_health = true
    name = "${aws_elb.lb.dns_name}"
    zone_id = "${aws_elb.lb.zone_id}"
  }
}
```

### Expected Behavior
The alias shouldn't be changed since nothing really changed

### Actual Behavior
The record gets replaced every time

**Comment by ColinHebert** [comment]
Probably relates to https://github.com/terraform-providers/terraform-provider-aws/issues/1700

**Comment by saxonww** [comment]
FWIW, changing `${aws_elb.lb.dns_name}` to `${lower(aws_elb.lb.dns_name)}` in the above example will work around this issue.

**Comment by bflad** [comment]
I have submitted a PR to hopefully fix this: #3119

**Comment by bflad** [comment]
The fix for this has been merged into master and will be released in v1.8.0 of the provider, expected tomorrow! 🎉

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.8.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#180-january-29-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
